src/backend/TrafficCourts/Citizen.Service/Features/Disputes/Create.cs

```diff
@@ -150,13 +163,14 @@ public async Task<Response> Handle(Request request, CancellationToken cancellati
 
                     SubmitNoticeOfDispute submitNoticeOfDispute = _mapper.Map<SubmitNoticeOfDispute>(dispute);
 
-                    Guid noticeOfDisputeGuid = NewId.NextSequentialGuid();
-                    submitNoticeOfDispute.NoticeOfDisputeGuid = noticeOfDisputeGuid;
+                    submitNoticeOfDispute.NoticeOfDisputeGuid = noticeOfDisputeId;
                     submitNoticeOfDispute.SubmittedTs = _clock.GetCurrentInstant().ToDateTimeUtc();
 
                     if (violationTicket != null)
                     {
-                        submitNoticeOfDispute.OcrTicketFilename = await _filePersistenceService.SaveJsonFileAsync(violationTicket, noticeOfDisputeGuid.ToString(), cancellationToken);
+                        var id = await SaveTicketOcrResultAsync(noticeOfDisputeId, violationTicket, cancellationToken);
+                        // OcrTicketFilename is obsolete, however, this field cannot be empty otherwise violation ticket will not be initialized
+                        submitNoticeOfDispute.OcrTicketFilename = id.ToString("d"); 

```

Comment:
So everything is `noticeOfDisputeId` based now as part of `InternalFileProperties` thus we don't need to store OcrTicketFilename. Nice!